File: app-gw/src/main/java/sdk/chat/demo/robot/api/model/GWConfigs.java
package sdk.chat.demo.robot.api.model;

import com.google.gson.annotations.SerializedName;

import java.util.List;
import java.util.Map;

public class GWConfigs {
    private GWConfigItem configs;

    @SerializedName("default_msg")
    private MessageDetail defaultMsg;

    @SerializedName("text_to_speech_voices")
    private List<DBVoiceType> dbVoiceTypes;


    @SerializedName("tts_voice")
    private Map<String, String> defaultVoiceTypes;

    @SerializedName("stt_config")
    private VoiceBaseConfigs voiceBaseConfigs;

    public Map<String, String> getDefaultVoiceTypes() {
        return defaultVoiceTypes;
    }

    public void setDefaultVoiceTypes(Map<String, String> defaultVoiceTypes) {
        this.defaultVoiceTypes = defaultVoiceTypes;
    }

    public VoiceBaseConfigs getVoiceBaseConfigs() {
        return voiceBaseConfigs;
    }

    public void setVoiceBaseConfigs(VoiceBaseConfigs voiceBaseConfigs) {
        this.voiceBaseConfigs = voiceBaseConfigs;
    }

    public GWConfigItem getConfigs() {
        return configs;
    }

    public void setConfigs(GWConfigItem configs) {
        this.configs = configs;
    }

    public List<DBVoiceType> getDbVoiceTypes() {
        return dbVoiceTypes;
    }

    public void setDbVoiceTypes(List<DBVoiceType> dbVoiceTypes) {
        this.dbVoiceTypes = dbVoiceTypes;
    }

//    public MessageDetail getWelcomeMsg() {
//        return welcomeMsg;
//    }
//
//    public void setWelcomeMsg(MessageDetail welcomeMsg) {
//        this.welcomeMsg = welcomeMsg;
//    }

    public static class GWConfigItem {
        @SerializedName("generating_hint")
        private List<String> generatingHint;

        public List<String> getGeneratingHint() {
            return generatingHint;
        }

        public void setGeneratingHint(List<String> generatingHint) {
            this.generatingHint = generatingHint;
        }
    }

    public MessageDetail getDefaultMsg() {
        return defaultMsg;
    }

    public void setDefaultMsg(MessageDetail defaultMsg) {
        this.defaultMsg = defaultMsg;
    }

    public static class VoiceBaseConfigs{
        private String token;
        @SerializedName("app_id")
        private String appId;

        public String getAppId() {
            return appId;
        }

        public void setAppId(String appId) {
            this.appId = appId;
        }

        public String getToken() {
            return token;
        }

        public void setToken(String token) {
            this.token = token;
        }
    }

    public static class DBVoiceType {
        private String name;
        @SerializedName("voice_type")
        private String voiceType;

        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }

        public String getVoiceType() {
            return voiceType;
        }

        public void setVoiceType(String voiceType) {
            this.voiceType = voiceType;
        }
    }
}

File: app-gw/src/main/java/sdk/chat/demo/robot/api/model/GWConfigs.kt
package sdk.chat.demo.robot.api.model

import com.google.gson.annotations.SerializedName

data class GWConfigs(
    @SerializedName("configs")
    val configs: GWConfigItem? = null,
    @SerializedName("contact_email")
    val contactEmail: String? = null,
    @SerializedName("privacy_agreement")
    val privacyAgreement: String? = null,
    @SerializedName("term_of_service")
    val termOfService: String? = null,

    @SerializedName("default_msg")
    val defaultMsg: MessageDetail? = null,

    @SerializedName("text_to_speech_voices")
    val dbVoiceTypes: List<DBVoiceType>? = null,

    @SerializedName("is_vip_display_crown")
    val isVipDisplayCrown: Boolean? = false,

    @SerializedName("tts_voice")
    val defaultVoiceTypes: Map<String, String>? = null,

    @SerializedName("stt_config")
    val voiceBaseConfigs: VoiceBaseConfigs? = null,

    @SerializedName("hymns_params")
    val hymnsParams: List<HymnParam>? = null,

    @SerializedName("welcome_survey")
    val welcomeSurvey: WelcomeSurvey? = null,
    @SerializedName("qr_code")
    val qrCode: QRCode? = null,
    val billingInfo: BillingInfo? = null

) {

    data class GWConfigItem(
        @SerializedName("generating_hint")
        val generatingHint: List<String>? = null
    )

    data class VoiceBaseConfigs(
        val token: String? = null,

        @SerializedName("app_id")
        val appId: String? = null
    )

    data class QRCode(
        val daily: String? = null,
        val verse: String? = null,
    )

    data class DBVoiceType(
        val name: String? = null,

        @SerializedName("voice_type")
        val voiceType: String? = null
    )

    data class HymnParam(
        val field: String? = null,
        val choices: List<String>? = null
    )

    data class WelcomeSurvey(
        val question: String? = null,
        val title: String? = null,
        val options: List<WelcomeSurveyOption>? = null,
        val background: String? = null,
        @SerializedName("event_page")
        val eventPage: String? = null,
    ) {
        fun findOptionByValue(value: String?): WelcomeSurveyOption? {
            if (value == null) return null
            return options?.find { it.value == value }
        }
    }

    data class WelcomeSurveyOption(
        val value: String? = null,
        val prompts: List<String>? = null,
        val response: String? = null
    )

    data class BillingInfo(
        val productSubscriptions: List<String>? = null
    )
}
